Makefile

```diff
@@ -362,7 +362,23 @@ switch-tanzu:
 #################################
 ##  --    Docker images    --  ##
 #################################
+# This regex will only match the below, without a trailing build/release candidate
+# 1. an optional preceding 'v'
+# 2. a standard semver '1.9.0'
+tag_regex_no_build_candidates := ^v{0,1}([0-9]{1,}\.){2}[0-9]{1,}$$
+
 docker-multiarch-build: go-generate generate-config-file 
+ifeq ($(shell echo ${IMG_VERSION} | egrep "${tag_regex_no_build_candidates}"),)
```

Comment:
Nice regexp :) I think we could otherwise rely on the environment variable [`SNAPSHOT` or `TAG_NAME`](https://github.com/elastic/cloud-on-k8s/blob/2efb002c1dde3ba63d6f039dc983b3dc8d24b32a/.ci/setenvconfig#L473-L510).

```
ifneq ($(SNAPSHOT),true) # check for `!true` so that if never it is not set, it's nightly
 // nightly
or
ifeq ($(strip $(TAG_NAME)),)
  // nightly
```

Reply:
Well @thbkrkr that shows my familiarity with the build process.  That's a lot cleaner.  Updated.  Thanks for the assist.